src/api/app/models/badge.rb

```diff
@@ -2,8 +2,9 @@ class Badge
   RED = '#e05d44'.freeze
   GREEN = '#4c1'.freeze
 
-  def initialize(type, results)
+  def initialize(type, buildresults)
     @type = 'unknown'
+    results = buildresults&.select { |r| ['succeeded', 'failed'].any?(r.code) }
```

Comment:
I would say this is a `state`. And `succeed` and `failed` are `finalstates`.

Reply:
According to [this line of code](https://github.com/openSUSE/open-build-service/blob/72e9bb9349a89b50815afb73d42221aba30bcb11/src/api/app/models/buildresult.rb#L68), these are considered final status: 'succeeded', 'failed', 'unresolvable', 'broken', 'disabled', 'excluded'

I can understand that we need to ignore "excluded" and "disabled" ones. However, I wonder if we should consider "unresolvable" or "broken" meaning "something wrong happened" as some action from the user is required to make it build. :thinking: